src/gridfs-stream/upload.ts

```diff
@@ -1,11 +1,43 @@
+import { WriteConcern } from './../write_concern';
 import * as crypto from 'crypto';
 import { PromiseProvider } from '../promise_provider';
 import { Writable } from 'stream';
 import { ObjectId } from '../bson';
-import type { Callback } from '../types';
+import type { Callback, AnyError } from '../types';
+import type { Collection } from '../collection';
+import type { Document } from './../types.d';
+import type { GridFSBucket } from './index';
+import type { GridFSFile } from './download';
+import type { WriteConcernOptions } from '../write_concern';
+import { MongoError } from '../error';
+import type { ObjectID } from 'bson';
 
 const ERROR_NAMESPACE_NOT_FOUND = 26;
 
+export type TFileId = string | number | object | ObjectId;
```

Comment:
Is there a convention for the leading `T` here?

Reply:
@mbroadst [referenced this spec](https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#user-content-terms:~:text=While,-GridFS) 

I'd be down to name it `GridFSFileId`, but I was assuming we'd need this in other places.

Reply:
@nbbeeken there is a convention for leading `T` on template parameters, so maybe that's where this originally came from. I don't have a strong preference either way